output/gigpress_sidebar.php

```diff
@@ -164,6 +166,25 @@ function form( $instance ) {
 			</select>
 		</p>
 
+		<p>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'related' ) ); ?>">
+				<?php _e( 'Only display shows related to this post', 'gigpress' ); ?>
+			</label>
+			<select style="width:100%;" id="<?php echo esc_attr( $this->get_field_id( 'related' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'related' ) ); ?>">
+				<option value="">--</option>
+				<?php
+				$posts = fetch_gigpress_related_posts();
+				if ( $posts != false ) :
+					foreach ( $posts as $this_post ) :
+						?>
+						<option value="<?php echo absint( $this_post->ID ); ?>" <?php echo ( $related == $this_post->ID ) ? 'selected="selected"' : ''; ?>>
```

Comment:
`absint( $this_post->ID )`

I wonder if [esc_attr](https://developer.wordpress.org/reference/functions/esc_attr/) would be a more appropriate escaping function here?

Probably doesn't matter as absint seems perfectly effective to me in this case, just leaving a note.

Reply:
When dealing with integers, absint and esc_attr are pretty interchangeable, with absint being the more restrictive one.